spec/micro/README.md

```diff
@@ -0,0 +1,770 @@
+# Micro IBC Architecture
+
+### Context
+
+The implementation of the entire IBC protocol as it currently stands is a large undertaking. While there exists ready-made implementations like ibc-go this is only deployable on the Cosmos-SDK. Similarly, there exists ibc-rs which is a library for chains to integrate. However, this requires the chain to be implemented in Rust, there still exists some non-trivial work to integrate the ibc-rs library into the target state machine, and certain limitations either in the state machine or in ibc-rs may prevent using the library for the target chain.
+
+Writing an implementation from scratch is a problem many ecosystems face as a major barrier for IBC adoption.
+
+The goal of this document is to serve as a "micro-IBC" specification that will allow new ecosystems to implement a protocol that can communicate with fully implemented IBC chains using the same security assumptions.
+
+The micro-IBC protocol must have the same security properties as IBC, and must be completely compatible with IBC applications. It may not have the full flexibility offered by standard IBC.
+
+### Desired Properties
+
+- Light-client backed security
+- Unique identifiers for each channel end
+- Authenticated application channel (must verify that the counterparty is running the correct client and app parameters)
+- Applications must be mutually compatible with standard IBC applications.
+- Must be capable of being implemented in smart contract environments with resource constraints and high gas costs.
+
+### Specification
+
+### Light Clients
+
+The light client module can be implemented exactly as-is with regards to its functionality. It **must** have external endpoints for relayers (off-chain processes that have full-node access to other chains in the network) to initialize a client, update the client, and submit misbehaviour in case the trust model of the client is violated by the counterparty consensus mechanism (e.g. committing to different headers for the same height).
+
+The implementation of each of these endpoints will be specific to the particular consensus mechanism targetted. The choice of consensus algorithm itself is arbitrary, it may be a Proof-of-Stake algorithm like CometBFT, or a multisig of trusted authorities, or a rollup that relies on an additional underlying client in order to verify its consensus.
```

Comment:
Probably requires single slot finality for the state proofs or some kind of finality gadget.

Reply:
I agree we could include some consideration around consensus finalities here.